### PR TITLE
Feature: Add Option to Expand Document Source

### PR DESCRIPTION
**Pull request by nung22**
### Description
_Gives users the ability to expand the row to view the entire document source with `OuiPopover`. Also refactors the `getDlTmpl` functional component by using `OuiDescriptionList`, `OuiDescriptionListTitle`, and `OuiDescriptionListDescription` in place of standard HTML tags. Removes the `:` in the source fields headers and updates their styling as well to make them more distinguishable from their values._

### Issues Resolved
_Closes #285_ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

# Screenshots

### Source details closed
issue-285-pic1

### Source details open
issue-285-pic2

### In action
issue-285-gif1

### CODE REVIEW

**Comment by kgcreative** [comment]
Nice incremental update here

File: public/components/query_compare/search_result/result_components/result_panel.tsx [hunk 1]
```diff
@@ -19,14 +19,21 @@ import { QueryError, SearchResults } from '../../../../types/index';
 import { useSearchRelevanceContext } from '../../../../contexts';
 
 import './result_components.scss';
+// import { set } from 'lodash';
```
Comment (sejli) [inline public/components/query_compare/search_result/result_components/result_panel.tsx:22]: Remove if unneeded

File: public/components/query_compare/search_result/result_components/result_grid.tsx [hunk 2]
```diff
@@ -27,19 +30,65 @@ interface ResultGridComponentProps {
   comparedDocumentsRank: DocumentRank;
   queryResult: SearchResults;
   resultNumber: number;
+  setQueryError: React.Dispatch<React.SetStateAction<QueryError>>;
```
Comment (sejli) [inline public/components/query_compare/search_result/result_components/result_grid.tsx:33]: Any reason why setting the `queryError` is required here? Seems like it's being set in `toggleDetails`, but I don't see how it's necessary.

Comment (nung22) [reply public/components/query_compare/search_result/result_components/result_grid.tsx:33]: Good catch, just removed it.

File: public/components/query_compare/search_result/result_components/result_panel.tsx (hunk 1 above)
Comment (nung22) [reply public/components/query_compare/search_result/result_components/result_panel.tsx:22]: Sounds good

File: public/components/query_compare/search_result/result_components/result_grid.tsx [hunk 3]
```diff
@@ -36,10 +39,51 @@ export const ResultGridComponent = ({
...
+        >
+          <EuiText
+            size="m"
+            className="eui-yScroll"
+            style={{
+              minWidth: 325,
+              width: '35vw',
+              wordWrap: 'break-word',
+              height: 220,
+              overflowY: 'auto',
```
Comment (sejli) [inline public/components/query_compare/search_result/result_components/result_grid.tsx:74]: Could we put this in the `scss` file?

**Review by sejli — COMMENTED** [review]
Great work, I love it! Left one nitpick comment, but may need to change the `.babelrc` as it's not passing tests still.

**Review by sejli — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
   },
   "dependencies": {},
   "devDependencies": {
+    "@babel/plugin-proposal-class-properties": "^7.18.6",
```
[the remark this answers is not in the document]
Comment (nung22) [reply package.json]: Thanks for catching that, just removed it!

**Review by nung22 — COMMENTED** [review]
Thanks for catching that! Just removed it.

**Review by sejli — APPROVED** [review]

**Pull request merged**